dulwich/repo.py

```diff
@@ -678,6 +678,25 @@ def __init__(self, root):
         self.hooks['commit-msg'] = CommitMsgShellHook(self.controldir())
         self.hooks['post-commit'] = PostCommitShellHook(self.controldir())
 
+    @classmethod
+    def discover(cls, start='.'):
+        """Iterate parent directories to discover a repository
+
+        Return a Repo object for the first parent directory that looks like a
+        Git repository.
+
+        :param start: The directory to start discovery from (defaults to '.')
+        """
+        path = os.path.abspath(start)
+        while path != '/':
```

Comment:
I'd suggest continuing until the second item in the tuple is empty. E.g.:

os.split.path("/") returns ("/", "")

This should be true on Windows as well.